extension/chrome/settings/setup.ts

```diff
@@ -488,6 +488,9 @@ Catch.try(async () => {
         return await Ui.modal.warning(e.message);
       } else if (e instanceof KeyCanBeFixed) {
         return await renderCompatibilityFixBlockAndFinalizeSetup(e.encrypted, options);
+      } else if (e instanceof ResetForm) {
```

Comment:
I know you followed the behavior of what we do in compose form here, with the reset exception. While that's what I did in composer, I'm not very proud of it, and don't really like this pattern. Anyway, in this case, there is no (large) harm in leaving the inputs as they are. I recommend to remove this if branch.

I will add related comments in other places.

Reply:
I think would be better to clean the form,
I can try to implement the same but in another way. What do you think?

Reply:
I think it's a fringe case, I don't mind not cleaning the form at all.

If the user is using an expired key, they are almost 100% a technical user. I don't mind to give them a bit of inconvenience having to press `CTRL+A` and then `CTRL+V` again with an updated key. They'll probably have to update the key in gnupg which is a lot more complicated.

Not worth complicating our codebase for this. However, if this was functionality that is expected to be used by normal clueless users, that would be a different story.

Reply:
Okay, Got it

Reply:
I just realized that I need to either replace this `ResetForm` with UserAlert or create a new one because I need to throw an exception here to stop validation here:
```TS
      } else if (await Pgp.key.usableButExpired(k)) {
        // Maybe would be better to give user 3 abilities:
        // 1) Confirm importing expired key
        // 2) Fix expired key
        // 3) Cancel
        const isConfirmed = await Ui.modal.confirm('You are importing a key that is expired. You can still import it to read messages from the past, ' +
          'but you will not be able to send new messages using this key. You can add more keys in the settings later.\n\nProceed with expired key?');
        if (!isConfirmed) {
          throw new ResetForm(); // <- here
        }
...
```

so maybe would be better to leave it as is or I can recode it but I don't really think that it's worth it

Reply:
Good point, you can `throw new UserAlert(` then, like a few lines below it does. At least it will be a bit more consistent, and the same logic can handle both situations.

Reply:
The UserAlert would then say: `You chose to not import expired key.\n\nPlease import another key, or edit the expired key in another OpenPGP software to extend key validity.`

Reply:
Okay!